packages/plugin-kit/src/fs/buildSliceDirectoryPath.ts

```diff
@@ -0,0 +1,26 @@
+import * as path from "node:path";
+
+import { pascalCase } from "./lib/pascalCase";
+import {
+	resolveSliceModel,
+	ResolveSliceModelArgs,
+} from "./lib/resolveSliceModel";
+
+import {
+	buildSliceLibraryDirectoryPath,
+	BuildSliceLibraryDirectoryPathArgs,
+} from "./buildSliceLibraryDirectoryPath";
+
+export type BuildSliceDirectoryPathArgs = BuildSliceLibraryDirectoryPathArgs &
+	ResolveSliceModelArgs;
+
+export async function buildSliceDirectoryPath(
+	args: BuildSliceDirectoryPathArgs,
+): Promise<string> {
+	const model = await resolveSliceModel(args);
```

Comment:
In most cases, we only provide a Slice ID to hooks, which means we need to load the model and derive a path using the Slice's name. We need to do this with Slices because their directory names are based on the name, not the ID.